airflow/providers/google/cloud/operators/bigquery.py

```diff
@@ -1692,32 +1692,52 @@ def prepare_template(self) -> None:
             with open(self.configuration, 'r') as file:
                 self.configuration = json.loads(file.read())
 
+    def _submit_job(self, hook: BigQueryHook, job_id: str):
+        # Submit a new job
+        job = hook.insert_job(
+            configuration=self.configuration,
+            project_id=self.project_id,
+            location=self.location,
+            job_id=job_id,
+        )
+        # Start the job and wait for it to complete and get the result.
+        job.result()
+        return job
+
     def execute(self, context: Any):
         hook = BigQueryHook(
             gcp_conn_id=self.gcp_conn_id,
             delegate_to=self.delegate_to,
         )
 
-        job_id = self.job_id or f"airflow_{self.task_id}_{int(time())}"
+        exec_date = context['execution_date'].isoformat()
+        job_id = self.job_id or f"airflow_{self.dag_id}_{self.task_id}_{exec_date}"
+
         try:
-            job = hook.insert_job(
-                configuration=self.configuration,
-                project_id=self.project_id,
-                location=self.location,
-                job_id=job_id,
-            )
-            # Start the job and wait for it to complete and get the result.
-            job.result()
+            # Submit a new job
+            job = self._submit_job(hook, job_id)
         except Conflict:
+            # If the job already exists retrieve it
             job = hook.get_job(
                 project_id=self.project_id,
                 location=self.location,
                 job_id=job_id,
             )
-            # Get existing job and wait for it to be ready
-            for time_to_wait in exponential_sleep_generator(initial=10, maximum=120):
-                sleep(time_to_wait)
-                job.reload()
-                if job.done():
-                    break
+
+            if job.done() and job.error_result:
+                # The job exists and finished with an error and we are probably reruning it
+                # So we have to make a new job_id because it has to be unique
+                job_id = f"{self.job_id}_{int(time())}"
+                job = self._submit_job(hook, job_id)
+            elif not job.done():
+                # The job is still running so wait for it to be ready
+                for time_to_wait in exponential_sleep_generator(initial=10, maximum=120):
```

Comment:
If you use the Google client you won't need to sleep and wait - the loop is built into the job itself

Reply:
Can you provide the code? I tried to use the `job.result()` but it doesn't work.

Reply:
```def bq_load(config: Config, client: bigquery.Client, event: Event, source_uri: str) -> JobResultEvent:
    logger.info("Loading GSheet to BQ")
    job_config = LoadJobConfig(
        autodetect=True,
        create_disposition="CREATE_IF_NEEDED",
        write_disposition="WRITE_TRUNCATE",
        source_format="NEWLINE_DELIMITED_JSON"
    )
    logger.info("loading_table")
    table_name = f"{config.project_id}.{config.dataset_name}.{event.destination_table}"
    job: LoadJob = client.load_table_from_uri("gs://" + source_uri, table_name, job_config=job_config)
    try:
        job.result()
        result_event = JobResultEvent.from_job_result_and_event(job, event)
    except (GoogleCloudError, TimeoutError, ClientError) as _:
        result_event = JobResultEvent.from_job_result_and_event(job, event)
        logger.error("BQ Job Failed")
    return result_event
``` That's one example of how I do it in an app

Reply:
When I hit result it polls too

Reply:
In case the job is already running I still do ```job.result()``` and it polls, but without ding reload